README.md

```diff
@@ -12,272 +16,131 @@ Install the gem
-
-```ruby
-user = client.user(
-  # Logged-in user id
-  id: "12345",
-
-  # Anonymous id
-  anonId: "abcdef",
+# Fetch features from GrowthBook API (or a Redis cache, etc.)
+uri = URI('https://cdn.growthbook.io/api/features/MY_API_KEY')
+res = Net::HTTP.get_response(uri)
+features = res.is_a?(Net::HTTPSuccess) ? JSON.parse(res.body)["features"] : nil
 
-  # Targeting attributes
+# Create a context for the current user/request
+gb = Growthbook::Context.new(
+  features: features,
+  # User attributes for targeting / variation assignment
   attributes: {
-    :premium => true,
-    :accountAge => 36,
-    :geo => [
-      :region => "NY"
-    ]
+    id: "123",
+    country: "US"
   }
-])
-```
+)
 
-You can update these at any time:
+# Use a boolean feature flag
+if gb.is_on? :my_feature_key
+  puts "My feature is on!"
+end
 
-```ruby
-user.id="54321"
-user.anonId = "fedcba"
-user.attributes={
-  :premium => false
-}
+# Get the value of a multivariate feature with a fallback
+btn_color = gb.feature_value(:signup_btn_color, "pink")
 ```
 
-## Experiment Configuration
+## Track Experiment Impressions
 
-The default test is a 50/50 split with no targeting or customization.  There are a few ways to configure this on a test-by-test basis.
+When a feature's value is determined by an experiment (A/B test), you typically want to track that assignment event for later analysis.
 
-### Option 1: Global Configuration
-
-With this option, you configure all experiments globally once and then reference them via id throughout the code.
+There are two ways to do this. First is by accessing all impressions at the end of a request:
 
 ```ruby
-client.experiments=[
-  # Default 50/50 2-way test
-  Growthbook::Experiment.new("my-test", 2),
-
-  # 3-way test with reduced coverage and unequal weights
-  Growthbook::Experiment.new(
-    "my-other-test", 
-    3, 
-    :coverage => 0.4, 
-    :weights => [0.5, 0.25, 0.25]
-  )
-]
-
-# Later in code, pass the string id instead of the Experiment object
-result = user.experiment("my-test")
+gb.impressions.each do |key, result|
+  puts "Assigned variation #{result.variation_id} in experiment #{key}"
+end
 ```
 
-There is a helper method that can import a hash of multiple experiments at once. The expected format follows the GrowthBook API response.
+Second is by using a listener to get alerted in realtime as users are put into experiments:
 
 ```ruby
-require 'json'
-
-# Example response from the GrowthBook API
-json = '{
-  "status": 200,
-  "experiments": {
-    "my-test": {
-      "variations": 2
-    },
-    "my-other-test": {
-      "variations": 3,
-      "weights": [0.5, 0.25, 0.25]
-    }
-  }
-}'
-parsed = JSON.parse(json)
+class MyImpressionListener
+  def on_experiment_viewed(experiment, result)
+    puts "Assigned variation #{result.variation_id} in experiment #{experiment.key}"
+  end
+end
 
-client.importExperimentsHash(parsed["experiments"])
+gb.listener = MyImpressionListener.new
```

Comment:
this sounds like a use case for ruby blocks/`yield`.
Instead of using a `MyImpressionListener.new`
, a ruby dev might prefer the this callback pattern from ActiveRecord:

<img width="809" alt="CleanShot 2022-05-18 at 17 51 13@2x" src="https://user-images.githubusercontent.com/55279023/169179193-ab04de37-9d9f-4731-9d74-98cc269732a2.png">